cluster/juju/layers/kubernetes-worker/actions.yaml

```diff
@@ -14,4 +14,30 @@ microbot:
       delete:
         type: boolean
         default: False
-        description: Removes a microbots deployment, service, and ingress if True.
+        description: Remove a microbots deployment, service, and ingress if True.
+registry:
+     description: Create a private Docker registry
+     params:
+      htpasswd:
+        type: string
+        description: base64 encoded htpasswd file used for authentication.
+      htpasswd-plain:
+        type: string
+        description: base64 encoded plaintext version of the htpasswd file, needed by docker daemons to authenticate to the registry.
+      tlscert:
```

Comment:
Which docker are we talking about? The one in the cluster, or the one on the user machine? The TLS interface will provide valid certs and the public ca trust cert to be added to ca-certificates. As such the docker/k8s bits will be able to verify the certs. As for the user I'd argue the action could return that public ca cert and instruct the user to add it to their chain.

For now I'll consider that a follow up.